src/Riok.Mapperly/Descriptors/Mappings/UserMappings/UserImplementedInlinedExpressionMapping.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Riok.Mapperly.Descriptors.Mappings.UserMappings;
+
+/// <summary>
+/// An inlined version of a <see cref="UserImplementedMethodMapping"/>.
+/// Does not support reference handling and has several other limitations,
+/// <see cref="InlineExpressionRewriter"/>.
+/// </summary>
+/// <param name="userMapping">The original user mapping.</param>
+/// <param name="sourceParameter">The source parameter of the user mapping. This will probably be rewritten when inlining.</param>
+/// <param name="mappingInvocations">Mapping invocations to be inlined.</param>
+/// <param name="mappingBody">The prepared user written mapping body code (rewritten by <see cref="InlineExpressionRewriter"/>.</param>
+public class UserImplementedInlinedExpressionMapping(
+    UserImplementedMethodMapping userMapping,
+    ParameterSyntax sourceParameter,
+    IReadOnlyDictionary<SyntaxAnnotation, INewInstanceMapping> mappingInvocations,
+    ExpressionSyntax mappingBody
+) : NewInstanceMapping(userMapping.SourceType, userMapping.TargetType), INewInstanceUserMapping
+{
+    public IMethodSymbol Method => userMapping.Method;
```

Comment:
This class contains lots of code that isn't covered by tests. Probably a good idea to add more test coverage